### ISSUE TITLE
Issue with context mode

### ISSUE THREAD

**Issue by tmarkiewicz** [psidnell/ofexport#1]
I see context mode is not documented, but it shows up in the command line help so I tired using it and got the following error:

$ ofexport -o /Users/tmarkiewicz/Documents/OmniFocus/index.html --tfi --tci none --tse 'from tomorrow' --prune -F -C
--tfi   = Task flagged includes flagged
Traceback (most recent call last):
  File "/Users/tmarkiewicz/scripts/ofexport/ofexport.py", line 262, in <module>
    traverse_list (visitor, items)
  File "/Users/tmarkiewicz/scripts/ofexport/treemodel.py", line 167, in traverse_list
    traverse (visitor, item, ignore_marked = ignore_marked)
  File "/Users/tmarkiewicz/scripts/ofexport/treemodel.py", line 173, in traverse
    traverse_context (visitor, item, ignore_marked = ignore_marked)
  File "/Users/tmarkiewicz/scripts/ofexport/treemodel.py", line 184, in traverse_context
    traverse_list (visitor, context.children, ignore_marked = ignore_marked)
  File "/Users/tmarkiewicz/scripts/ofexport/treemodel.py", line 167, in traverse_list
    traverse (visitor, item, ignore_marked = ignore_marked)
  File "/Users/tmarkiewicz/scripts/ofexport/treemodel.py", line 177, in traverse
    traverse_task (visitor, item, ignore_marked = ignore_marked)
  File "/Users/tmarkiewicz/scripts/ofexport/treemodel.py", line 190, in traverse_task
    visitor.begin_any (task)
  File "/Users/tmarkiewicz/scripts/ofexport/visitors.py", line 69, in begin_any
    item.attribs[SELECTED] = item.parent.attribs[SELECTED]
KeyError: 'visitors.selected'

My use case is showing a flagged view of tasks that aren't in the future. It works great in the regular project mode though (thanks!!).

**Comment by psidnell** [comment]
Hi, thanks for the feedback!

Yes its pretty broken which I haven't documented it yet :-) I've been trying to get project mode and the date filtering robust first - nearly there.

There's a fundamental design problem with filters and context mode at the moment:

My filter implementation makes use of the parent/child relationship of the tasks (the same structure as in the OF database) when it's traversing. a tasks parent is it's parent task/project. A projects parent (if it has one) is it's parent folder and a folders parent (if it has one) is it's parent folder, and so on - a nice basic tree.

Unfortunately the context structure isn't quite as straight forward. I've modelled the context hierarchy OK but the way tasks relate to it cause some issues where things get processed twice breaking the strict order I need to make it work.

Since a project AND its tasks are directly IN a context then when I walk a context and process those items in it I'll walk into the project and traverse it and later separately into the individual tasks.

I can see how I might do it which a second kind of filter but I want to try and re-use the existing filters if possible.

I've got a big re-design of filters coming in 1.0.2 (I made a design boo boo - try filtering on a 2nd level folder, oops) that I hope to get out this weekend, and then I'm going to think about context mode in 1.0.3.

The TODO.md doc should give you a clue about what's coming.

More information than you require :-)

Thanks again.
